src/cpp/elf/aie2ps/aie2ps_elfwriter.h

```diff
@@ -10,12 +10,71 @@ namespace aiebu {
 
 class aie2ps_elf_writer: public elf_writer
 {
-  constexpr static unsigned char ob_abi = 0x40;
+  constexpr static unsigned char ob_abi = 0x46;
   constexpr static unsigned char version = 0x02;
 public:
   aie2ps_elf_writer(): elf_writer(ob_abi, version)
   { }
 };
 
+class aie2ps_config_elf_writer: public elf_writer
+{
+  constexpr static unsigned char ob_abi = 0x46;
+  constexpr static unsigned char version = 0x03;
+  const std::string const_configuration = "configuration";
+  const std::string xrt_configuration = ".note.xrt.configuration";
+  const std::string const_kernel_signature = "kernel.signature";
+
+public:
+  aie2ps_config_elf_writer(): elf_writer(ob_abi, version)
+  { }
+
+  void
+  add_group(std::string name, std::vector<uint32_t> member, ELFIO::Elf_Word info_index)
+  {
+    // add section
+    ELFIO::section* sec = m_elfio.sections.add(name);
+    sec->set_type(ELFIO::SHT_GROUP);
+    sec->set_flags(ELFIO::SHF_ALLOC);
+    sec->set_addr_align(align);
+    sec->set_info(info_index);
+    sec->set_entry_size(4);
+
+    if(member.size())
+      sec->set_data(reinterpret_cast<const char*>(member.data()), static_cast<ELFIO::Elf_Word>(member.size()*4));
+
+    const ELFIO::section* lsec = m_elfio.sections[".symtab"];
+    sec->set_link(lsec->get_index());
+  }
+
+  std::vector<char>
+  process(std::vector<std::shared_ptr<writer>>& mwriter) override
+  {
+    auto mconfig_writer = std::dynamic_pointer_cast<config_writer>(mwriter[0]);
+    init_symtab();
+    int index=0;
+    for( auto& [kernel, instances] : mconfig_writer->get_kernel_map())
+    {
+       auto kernel_index = add_symtab(kernel);
+       for(auto& [iname, instance] : instances)
+       {
+         auto instance_index = add_symtab_section(iname, kernel_index);
+         std::vector<uint32_t> group_data = process_common_helper(instance, "."+std::to_string(index));
```

Comment:
As in other parts of the code base, please use helper functions when generating names.